client/src/main/kotlin/com/lowes/auditor/client/infrastructure/frameworks/mapper/JsonNodeMapper.kt

```diff
@@ -26,77 +26,65 @@ object JsonNodeMapper {
-                                            if (eventType == EventType.DELETED) {
-                                                findType(t.t2)?.let { it1 -> getValue(it1, t.t2) }
-                                            } else {
-                                                null
-                                            }
-                                        Flux.just(
-                                            Element(
-                                                name = entry.key,
-                                                updatedValue = updatedValue,
-                                                previousValue = previousValue,
-                                                metadata =
-                                                    ElementMetadata(
-                                                        fqdn = fqcnValue,
-                                                    ),
-                                            ),
-                                        )
-                                    } else {
-                                        toElement(t.t2, eventType, fqcnValue)
-                                    }
-                                }
-                        } else if (it == NodeType.OBJECT) {
-                            toElement(entry.value, eventType, getFqcnValue(hasFields, index, fqcn, entry))
-                        } else {
-                            val updatedValue =
-                                if (eventType == EventType.CREATED) {
-                                    getValue(it, entry.value)
-                                } else {
-                                    null
-                                }
-                            val previousValue =
-                                if (eventType == EventType.DELETED) {
-                                    getValue(it, entry.value)
-                                } else {
-                                    null
-                                }
-                            Flux.just(
-                                Element(
-                                    name = entry.key,
-                                    updatedValue = updatedValue,
-                                    previousValue = previousValue,
-                                    metadata =
-                                        ElementMetadata(
-                                            fqdn = getFqcnValue(hasFields, index, fqcn, entry),
-                                        ),
-                                ),
-                            )
-                        }
-                    } ?: Flux.empty()
+            .flatMap { tuple ->
+                val index = tuple.t1
+                val entry = tuple.t2
+                val elementFqcn = getFqcnValue(hasFields, index, fqcn, entry)
+                val nodeType = findType(entry.value) ?: return@flatMap Flux.empty<Element>()
```

Comment:
Lets avoid using non local return, e.g return@flatMap,
too many returns/jumps in code blocks reduces the readability as code grows.. Rather i would suggest a traditional if/else or ?.let calls as you prefer.